src/amo/reducers/versions.js

```diff
@@ -101,12 +123,23 @@ type GetBySlugParams = {|
 export const getVersionsBySlug = ({
   slug,
   state,
-}: GetBySlugParams): Array<AddonVersionType> | null => {
+}: GetBySlugParams): Array<AddonVersionType> => {
   invariant(slug, 'slug is required');
   invariant(state, 'state is required');
 
   const infoForSlug = state.bySlug[slug];
-  return (infoForSlug && infoForSlug.versions) || null;
+  if (infoForSlug) {
+    return infoForSlug.versionIds.map((versionId) => {
+      const version = getVersionById({ id: versionId, state });
+      if (!version) {
```

Comment:
I didn't realize that `loadAddons` is guaranteed to [load a version object](https://github.com/mozilla/addons-frontend/blob/e6b1a6d90bec1ddae26ae360fff074c59953a499/src/amo/reducers/versions.js#L221-L224). Because of that, you don't need to throw an error. 

However, I think it would be nice to add `invariant(version, ...)` in case someone refactors the `loadAddons` reducer and accidentally removes the `byId` part. Calls to `invariant()` are always executed which means you don't need to add test coverage for them -- there is no chance of introducing unexpected errors.